code/src/App.js

```diff
@@ -1,9 +1,98 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+
+import Header from 'components/Header';
+import NewPost from 'components/NewPost';
+import PostList from 'components/PostList';
 
 export const App = () => {
+  const [postList, setPostList] = useState([]);
+  const [newPost, setNewPost] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const fetchData = () => {
+    setLoading(true);
+    fetch('https://happy-thoughts-technigo.herokuapp.com/thoughts')
+      .then((data) => data.json())
+      .then((transformedData) => setPostList(transformedData))
+      .catch((error) => console.error(error))
+      .finally(() => setLoading(false));
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  const onNewPostChange = (event) => {
+    setNewPost(event.target.value);
+  };
+
+  const handleFormCleanup = () => {
+    setNewPost('');
+    setLoading(false);
+  };
+
+  const handleFormSubmit = (event) => {
+    event.preventDefault();
+    const options = {
+      method: 'post',
+      body: JSON.stringify({
+        message: newPost
+      }),
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    };
+    setLoading(true);
+    fetch('https://happy-thoughts-technigo.herokuapp.com/thoughts', options)
+      .then((data) => data.json())
+      .then(() => fetchData())
+      .catch((error) => console.error(error))
+      .finally(() => handleFormCleanup());
+  };
+
+  if (loading) {
```

Comment:
Great that you added a loading, it shows up when the page is first loading and doesn't glitch when a thought is posted, nice!